test/e2e/network/service.go

```diff
@@ -2679,6 +2681,118 @@ var _ = SIGDescribe("Services", func() {
 
 		framework.ExpectEqual(foundSvc, true, "could not find service 'kubernetes' in service list in all namespaces")
 	})
+
+	ginkgo.It("should test the lifecycle of an Endpoint", func() {
+		ns := f.Namespace.Name
+		testEndpointName := "testservice"
+
+		ginkgo.By("creating an Endpoint")
+		_, err := f.ClientSet.CoreV1().Endpoints(ns).Create(context.TODO(), &v1.Endpoints{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      testEndpointName,
+				Namespace: ns,
+				Labels: map[string]string{
+					"testendpoint-static": "true",
+				},
+			},
+			Subsets: []v1.EndpointSubset{{
+				Addresses: []v1.EndpointAddress{{
+					IP: "10.0.0.24",
+				}},
+				Ports: []v1.EndpointPort{{
+					Name:     "http",
+					Port:     80,
+					Protocol: v1.ProtocolTCP,
+				}},
+			}},
+		}, metav1.CreateOptions{})
+		framework.ExpectNoError(err, "failed to create Endpoint")
+
+		// setup a watch for the Endpoint
+		endpointWatchTimeoutSeconds := int64(60)
```

Comment:
To be clear, this is for the *entire* life of the watch / test. Is this in line with expected runtime for this test under heavily loaded system? Otherwise it could flake. We would know based on looking at timeout values for other tests - we may also want to re-use an existing constant.